modules/core/src/matrix.cpp

```diff
@@ -1656,7 +1656,7 @@ Size _InputArray::size(int i) const
     {
         const Mat* vv = (const Mat*)obj;
         if( i < 0 )
-            return sz.height==0 ? Size() : Size(sz.height, 1);
+            return sz.height==0 ? Size() : Size(1, sz.height);
```

Comment:
This fix is not consistent with another code. Check for `STD_VECTOR_MAT` or `STD_VECTOR_VECTOR` few lines above.

BTW, The most part of this code is not [covered](http://pullrequest.opencv.org/buildbot/export/opencv_releases/master_coverage-lin64-debug/20180117-000112_10342/coverage_html/opencv/modules/core/src/matrix.cpp.gcov.html) by tests (`std::array` requires C++11)

Reply:
Because the current implementation considers `std::vector<cv::Mat>` as a **row** of matrices, while `std::array<cv::Mat, N>` is treated as a **column** of matrices.